src/seq2seq.ipynb

```diff
@@ -0,0 +1,686 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 56,
+   "id": "initial_id",
+   "metadata": {
+    "collapsed": true,
+    "ExecuteTime": {
+     "end_time": "2024-03-23T09:39:36.799678Z",
+     "start_time": "2024-03-23T09:39:36.796488Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "import torch.optim as optim\n",
+    "import torch.nn.functional as F\n",
+    "from torch.utils.data import TensorDataset, RandomSampler, DataLoader\n",
+    "import numpy as np\n",
+    "import random\n",
+    "\n",
+    "from tqdm import tqdm\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "outputs": [],
+   "source": [
+    "random.seed(42)\n",
+    "np.random.seed(42)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-03-23T09:13:14.631620Z",
+     "start_time": "2024-03-23T09:13:14.627465Z"
+    }
+   },
+   "id": "cd68bec60e2d33c0",
+   "execution_count": 2
+  },
+  {
+   "cell_type": "code",
+   "outputs": [],
+   "source": [
+    "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-03-23T09:13:14.636571Z",
+     "start_time": "2024-03-23T09:13:14.633144Z"
+    }
+   },
+   "id": "ba66ac064b0e256d",
+   "execution_count": 3
+  },
+  {
+   "cell_type": "code",
+   "outputs": [],
+   "source": [
+    "# dependent on the implementation of the word2index and index2word\n",
+    "SOS_token = 0\n",
+    "EOS_token = 1\n",
+    "MAX_LENGTH = 80 + 2 # 80 words + SOS and EOS"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-03-23T09:33:11.689003Z",
+     "start_time": "2024-03-23T09:33:11.677603Z"
+    }
+   },
+   "id": "82cf8dea8a0ef552",
+   "execution_count": 38
+  },
+  {
+   "cell_type": "code",
+   "outputs": [],
+   "source": [
+    "class Lang:\n",
+    "    def __init__(self, name):\n",
+    "        self.name = name\n",
+    "        self.word2index = {}\n",
+    "        self.word2count = {}\n",
+    "        self.index2word = {0: \"SOS\", 1: \"EOS\"}\n",
+    "        self.n_words = 2  # Count SOS and EOS\n",
+    "\n",
+    "    def addSentence(self, sentence):\n",
+    "        for word in sentence.split(' '):\n",
+    "            self.addWord(word)\n",
+    "\n",
+    "    def addWord(self, word):\n",
+    "        if word not in self.word2index:\n",
+    "            self.word2index[word] = self.n_words\n",
+    "            self.word2count[word] = 1\n",
```

Comment:
then wouldn't each unique word have value 1? since this is an assignment, not an increment

Reply:
it is an assignment for words that are not yet in the dictionary. We increment for words that are already there

Reply:
oooh alrighty